### ISSUE TITLE
False or no response 

### ISSUE THREAD

**Issue by MjacobG** [Vsaggiomo/Ender3-syringe-pumps#4]
Hello! Thank you very much for your tutorial, this is an awesome project! I am trying to replicate the project using one stepper motor.  The motherboard I am using is the Creality3D  V1.1.4 which belongs to my old Ender 3 Pro. The firmware of the V1.1.4 has been upgraded to the "TH3D U1.R2.A8". The stepper motor I use is a Creality 40-42 with official stepper drivers . The lead screw is 8mm in diameter and has 2mm thread Pitch (looks like the original z axis).

I tried to run your gcode, slightly modified like this: 

Line1: ; Experiment 01 27/01/2023
Line2: ; 5 mL syringe, one pump
Line3: ; ID of 5 mL syringe is 12.4 mm
Line4:
Line5: M92 Z-33123 ; set the proper steps/mL for the 5mL syringe!
Line6:
Line7: M302 S0 ; print without checking temperature 
Line8: M211 S1 ; print with checking the Z end stop 
Line9: G91 ; relative positioning 
Line10: ; deliver 1mL at 1mL/min sequentially 
Line11: G1 Z1 F1 
Line12: ; wait for 10 seconds
Line13: G4 S10
Line14: ; reverse movement
Line15: G1 Z-1 F1 

Result: 
1. The pusher block starts immediately and moves backwards until it hits the Z end stop. Then, after 10 seconds, the pusher block moves forward (towards the syringe, that's good) for 1 minute covering a distance of 8.7 cm. Exceeds the length of a 5ml syringe. 
2. When Line11 is inactive ( ;G1 Z1 F1 ), nothing happens
3. When Line15 is inactive ( ;G1 Z-1 F1 ), the gcode stopes after the push block hits the z end stop.
4.  The 33123 value, obtained from your excel file after replacing the ID of my 5ml syringe. 

I am confused because I would like to simple have a flow of 1ml in 1ml/min and the results are not even close to the ideal scenario.

I would appreciate any valuable feedback! 
Thank you in advance, 
Jacob

**Comment by Vsaggiomo** [comment]
Hi there, I see multiple problems there :) 

1) First of all you are using another firmware and I'm not sure if it's going to work. It should, but you never know. 

2) You are using a screw that is different from what we use. As the linear motion is dictated by the pitch of the screw, the M92 command is clearly wrong. You can calculate it here https://blog.prusa3d.com/calculator_3416/ if it's a standard screw. Or you can just put a number in the M92 and try to move the axis of 5cm and adjust that number until you reach exactly a motion of 5mc. 

3) Most probably you are running a faulty gcode on the sd card. There should be no empty line and it should be written with a proper text editor, for example notepad++ using EOL (end of line) from Unix. I would start from proterface, using a USB cable and check with the computer.

**Comment by MjacobG** [comment]
Thank you very much for your prompt response! In fact the solution was very easy and right in front of my eyes! After your 2nd commend I just realized that I use an identical copy of the original z-axis lead screw. So, I deleted the M92 code and using the cylinder volume function, I solved for l (length) and I found that in order to dispense 1ml, the push block has to move 8.15mm. So the G-code Was just one line code: 
G1 Z8.15 F1 
Know I have to calculate the right F speed. Thank you very for your comments :)
